lib/registrations/registration_checker.rb

```diff
@@ -129,10 +129,18 @@ def validate_qualifications!(request, competition, target_user)
       end
 
       def validate_guests!(guests, competition)
-        raise WcaExceptions::RegistrationError.new(:unprocessable_entity, ErrorCodes::INVALID_REQUEST_DATA) if guests < 0
-        raise WcaExceptions::RegistrationError.new(:unprocessable_entity, ErrorCodes::GUEST_LIMIT_EXCEEDED) if competition.guest_limit_exceeded?(guests)
-        raise WcaExceptions::RegistrationError.new(:unprocessable_entity, ErrorCodes::UNREASONABLE_GUEST_COUNT) if
-          guests > DEFAULT_GUEST_LIMIT && !competition.guest_entry_status_restricted?
+        r = Registration.new(guests: guests, competition: competition)
```

Comment:
The target idea here is that we do **not** have to create an "on-the-fly" registration in every `validate_*` method. Rather, we should create the registration exactly as the user/request wishes but without persisting it to the DB, and then "pass it around" before calling the final `save`.

In other words, in an ideal world this `r` Registration already exists and can be passed into the method. Right now, it's okay to create on the fly because this PR is more of a proof-of-concept.